### PR TITLE
[Debt] Removes Record of Decision feature flag

### LINKED ISSUE #8415: ♻️ Remove `FEATURE_RECORD_OF_DECISION` flag
## ♻️ Debt/Refactor

Remove the `FEATURE_RECORD_OF_DECISION` flag some time after it has been enabled.

## ✅ Acceptance Criteria

A set of assumptions which, when tested, verify that the debt was addressed and expected functionality has not been affected.

- [ ] Flag removed from codebase
- [ ] Flag removed from IaC
- [ ] Remove debt code (eg. RODViewPoolCandidatePage.tsx)

## 🛑 Blockers
```[tasklist]
### Blocked By
- [ ] #8834
```

### PR DESCRIPTION
**Pull request by mnigh**
🤖 Resolves #8415.

## 👋 Introduction

This PR removes the Record of Decision feature flag. It also [adds `PoolTestSeeder` to `CiSeeder`](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9879/commits/2581bdddc06b0b3adda7801b94525b4d4b5bd3ea) so that the **pools** Cypress test has pools to test against.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Verify app has no discernible differences

## 🚚 Deployment

Add any additional details that are required for deploying the application.

Remove `FEATURE_RECORD_OF_DECISION` env var.

### CODE REVIEW

File: apps/e2e/cypress/e2e/admin/pool-candidate.cy.ts
Comment (mnigh) [inline apps/e2e/cypress/e2e/admin/pool-candidate.cy.ts:1]: This file could be entirely removed since it is testing a page that no longer exists, but this is out of scope for this issue. Creating new tests might be a good idea but should probably be done in Playwright and not Cypress.

File: apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/CareerTimelineSection/CareerTimelineSection.tsx [hunk 1]
```diff
@@ -36,15 +33,6 @@ const CareerTimelineSection = ({ experiences }: CareerTimelineSectionProps) => {
...
     <>
-      {features.recordOfDecision && (
-        <Heading Icon={UserCircleIcon} color="tertiary">
-          {intl.formatMessage({
-            defaultMessage: "Career timeline",
-            id: "5FK/4R",
-            description: "Heading for the users experiences",
-          })}
-        </Heading>
-      )}
```
Comment (esizer) [inline apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/CareerTimelineSection/CareerTimelineSection.tsx:47]: Looks like the opposite happened :sweat_smile:

**Review by esizer — CHANGES_REQUESTED** [review]
Wow, this really seeped into a lot of places :stuck_out_tongue: 

One minor thing but otherwise, this looks good!

File: apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/CareerTimelineSection/CareerTimelineSection.tsx (hunk 1 above)
Comment (mnigh) [reply apps/web/src/pages/PoolCandidates/ViewPoolCandidatePage/components/CareerTimelineSection/CareerTimelineSection.tsx:47]: Whoops.  1c87e20e2e86f4626934368f908bbf7d812c85a2

File: apps/playwright/tests/applicant/record-of-decision.spec.ts
[the remark this answers is not in the document]
Comment (mnigh) [reply apps/playwright/tests/applicant/record-of-decision.spec.ts:1]: Yes, documentation seems appropriate. Maybe `documentation/environment-variables.md`?

[the remark this answers is not in the document]
Comment (esizer) [reply apps/playwright/tests/applicant/record-of-decision.spec.ts:1]: oh wait...[we already do](https://github.com/GCTC-NTGC/gc-digital-talent/blob/main/documentation/environment-variables.md#override-for-entire-spec) :laughing:

**Review by esizer — APPROVED** [review]
LGTM! :+1:
